src/test/java/com/fewlaps/quitnowemailsuggester/StringUtilsTest.java

```diff
@@ -34,4 +37,53 @@ public void shouldReturnHeo() {
     public void shouldReturnAaabxx() {
         assertEquals("aaabxx", su.replaceLast("aaabbb", "bb", "xx"));
     }
+
+
+    @Test
+    public void shouldReturnNullForNull() throws InvalidEmailException {
+        String[] parts = su.getParts(null);
+        assertNull(parts);
+    }
+
+    @Test
+    public void shouldReturnArrayWithxZeroElementsFromEmpty() throws InvalidEmailException {
+        String[] elements = su.getParts("");
+
+        Assert.assertEquals(elements.length, 0);
+    }
+
+    @Test
+    public void shouldReturnArrayWithOneElementFromSimpleEmail() throws InvalidEmailException {
+        String[] elements = su.getParts("bernat.borras@example.com");
+
+        Assert.assertEquals(elements.length, 1);
+    }
+
+    @Test
+    public void shouldReturnArrayWithTwoElementFromTwoEmails_whenNullSeparator() throws InvalidEmailException {
+        String[] elements = su.getParts("bernat.borras@example.com:Office365", null);
+
+        Assert.assertEquals(elements.length, 2);
+    }
+
+    @Test
+    public void shouldReturnArrayWithTwoElementFromTwoEmails_whenEmptySeparator() throws InvalidEmailException {
+        String[] elements = su.getParts("bernat.borras@example.com:Office365", "");
+
+        Assert.assertEquals(elements.length, 2);
+    }
+
+    @Test
+    public void shouldReturnArrayWithTwoElementFromTwoEmails() throws InvalidEmailException {
+        String[] elements = su.getParts("bernat.borras@example.com:Office365");
+
+        Assert.assertEquals(elements.length, 2);
+    }
+
+    @Test
+    public void shouldReturnArrayWithTwoElementFromTwoPartsWithColon() throws InvalidEmailException {
+        String[] elements = su.getParts("bernat.borras@example.com,Office365", ",");
```

Comment:
This is something that will happen in the real world? Looking at *EmailSuggester*, it seems that the *separator* value always has the same value. Why not simply removing it? It will generate less code, less bugs and less tests.